src/modules/powerrename/PowerRenameUILib/MainWindow.xaml.cpp

```diff
@@ -273,6 +273,8 @@ namespace winrt::PowerRenameUI::implementation
         {
             ToggleAll();
             m_allSelected = !m_allSelected;
+            m_explorerItems.Clear();
+            PopulateExplorerItems();
```

Comment:
this change doesn't make sense to me. On SelectAll/UnsellectAll items list remains the same, so re-population is not needed. Only the change in `ToggleAll()` should be enough here

Reply:
With the filter, what is not being renamed should not be shown, at least that is what I understood. That's why I use the re-population.
If you unselect all and it is set to "Only show files that will be renamed", should we still show anything in the list?
with the change in ToggleAll() it stops the crash, but the list shown did not look right to me.

Reply:
ok. that makes sense. As PopulateExplorerItems() is heavyweight function, I'd suggest to call this only when filter is ShowRenamed. Also don't forget to clear explorerItemsMap as well and UpdateCounts()